api/client/client_test.go

```diff
@@ -41,35 +41,129 @@ func serverFunc(api *api.API) swarmhttp.TestServer {
 
 // TestClientUploadDownloadRaw test uploading and downloading raw data to swarm
 func TestClientUploadDownloadRaw(t *testing.T) {
-	testClientUploadDownloadRaw(false, t)
+	srv := swarmhttp.NewTestSwarmServer(t, serverFunc, nil, nil)
+	defer srv.Close()
+
+	data := []byte("foo123")
+	testClientUploadDownloadRaw(srv, false, t, data, false)
+
+	// check the tag was created successfully
+	tag := srv.Tags.All()[0]
+	testutil.CheckTag(t, tag, 1, 1, 0, 1)
 }
 
 func TestClientUploadDownloadRawEncrypted(t *testing.T) {
+
 	if testutil.RaceEnabled {
 		t.Skip("flaky with -race on Travis")
 		// See: https://github.com/ethersphere/go-ethereum/issues/1254
 	}
 
-	testClientUploadDownloadRaw(true, t)
+	srv := swarmhttp.NewTestSwarmServer(t, serverFunc, nil, nil)
+	defer srv.Close()
+
+	data := []byte("foo123")
+	testClientUploadDownloadRaw(srv, true, t, data, false)
+
+	// check the tag was created successfully
+	tag := srv.Tags.All()[0]
+	testutil.CheckTag(t, tag, 1, 1, 0, 1)
+}
+
+func TestPinWithRawUpload(t *testing.T) {
+	srv := swarmhttp.NewTestSwarmServer(t, serverFunc, nil, nil)
+	defer srv.Close()
+
+	data := testutil.RandomBytes(1, 10000)
+	hash := testClientUploadDownloadRaw(srv, false, t, data, true)
+
+	// Check if the file is pinned
+	CheckIfPinned(t, srv, hash, nil, 1, true)
```

Comment:
`CheckIfPinned` and `CheckIfUnpinned` should not be exported

Reply:
Done